### PR TITLE
workflow update to mail error log w/signoff

### PR DESCRIPTION
**Pull request by Jougan-0**
**Notes for Reviewers**

Update the workflow to send error logs whenever an error occurs while running mesheryctl registry generate. Additionally, create a new file to store only the errors.

Local run of the [workflow](https://github.com/meshery/meshery/pull/11156).

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

File: mesheryctl/internal/cli/root/registry/generate.go [hunk 1]
```diff
@@ -417,6 +425,16 @@ func logModelGenerationSummary(modelToCompGenerateTracker *store.GenerticThreadS
 	utils.Log.Info(fmt.Sprintf("-----------------------------\n-----------------------------\nGenerated %d models and %d components", totalAggregateModel, totalAggregateComponents))
 }
 
+func logError(err error, modelName string) {
```
Comment (MUzairS15) [inline mesheryctl/internal/cli/root/registry/generate.go:428]: No need.
Use log instance,  refer the existing approach which logs its output to registry log file.
Create a new log instance for errors logs when an error occurs use that log instance. As you initialize that instance set its log output to registry-generate-error.

Also make the directory structured, if you would.
`logs/registry/1. log (contains logs of al severity except error logs) 2. `error (contains only error logs)`

**Review by MUzairS15 — CHANGES_REQUESTED** [review]

File: mesheryctl/internal/cli/root/registry/generate.go (hunk 1 above)
Comment (Jougan-0) [reply mesheryctl/internal/cli/root/registry/generate.go:428]: Will do so positively.

**Review by leecalcote — COMMENTED** [review]
Doe this new email replace the existing email for this workflow or will it be a duplicate?

**Comment by Jougan-0** [comment]
> Does this new email replace the existing email for this workflow or will it be a duplicate?

This will replace the component generator failure email in these scenarios:

- >     If the component generator job fails, the old failure email will be used.
- >     If the component generator is successful but the error log is not empty, a new email regarding the error logs will be sent, and the workflow will be marked as successful.

For the scenario if the generator works and updater fails.
- >   We will have two emails: one for the failure of the updater and one for the error logs.

**Review by leecalcote — APPROVED** [review]
LGTM

**Pull request merged**